silver/api/views.py

```diff
@@ -941,3 +941,35 @@ def get_object(self):
             raise Http404
 
         return get_object_or_404(Transaction, uuid=uuid)
+
+
+class TransactionAction(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+        transaction = self.get_object(**kwargs)
+        action_to_execute = getattr(transaction, kwargs.get('requested_action'),
+                                    None)
+
+        if not action_to_execute:
```

Comment:
Adding a list of allowed actions would be safer. The URL can still keep it's regex although it wouldn't be necessarily needed anymore. This is valid for the `PaymentMethodAction` view as well.